File: Core-BackEnd/Investa.Domain/Entities/InvestmentImage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Investa.Domain.Entities.Enums;

namespace Investa.Domain.Entities;

public class InvestmentImage
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int InvestmentId { get; set; }

    /// <summary>
    /// Type of media asset (CoverImage, Image, Video)
    /// </summary>
    [Required]
    public MediaType MediaType { get; set; } = MediaType.Image;

    [Required]
    [StringLength(500)]
    public string Url { get; set; } = null!;

    /// <summary>
    /// Thumbnail URL for videos or optimized image thumbnails
    /// </summary>
    [StringLength(500)]
    public string? ThumbnailUrl { get; set; }

    /// <summary>
    /// Original file name
    /// </summary>
    [StringLength(255)]
    public string? FileName { get; set; }

    [StringLength(250)]
    public string? Caption { get; set; }

    public int SortOrder { get; set; } = 0;

    public bool IsPrimary { get; set; } = false;

    /// <summary>
    /// User who uploaded this media asset
    /// </summary>
    public Guid? UploadedBy { get; set; }

    public DateTime CreatedAt { get; set; }

    [ForeignKey(nameof(InvestmentId))]
    public Investment Investment { get; set; } = null!;

    [ForeignKey(nameof(UploadedBy))]
    public AuthUser? Uploader { get; set; }
}

File: Flutter_Partner/lib/models/investment_image.dart
/// Media type for investment images/videos
enum MediaType { coverImage, image, video }

class InvestmentImage {
  final int id;
  final int mediaType;
  final String url;
  final String? thumbnailUrl;
  final String? fileName;
  final String? caption;
  final int sortOrder;
  final bool isPrimary;
  final String? uploadedBy;
  final DateTime createdAt;

  InvestmentImage({
    required this.id,
    required this.mediaType,
    required this.url,
    this.thumbnailUrl,
    this.fileName,
    this.caption,
    required this.sortOrder,
    required this.isPrimary,
    this.uploadedBy,
    required this.createdAt,
  });

  factory InvestmentImage.fromJson(Map<String, dynamic> json) {
    return InvestmentImage(
      id: json['id'],
      mediaType: json['mediaType'] ?? 1,
      url: json['url'],
      thumbnailUrl: json['thumbnailUrl'],
      fileName: json['fileName'],
      caption: json['caption'],
      sortOrder: json['sortOrder'] ?? 0,
      isPrimary: json['isPrimary'] ?? false,
      uploadedBy: json['uploadedBy']?.toString(),
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'mediaType': mediaType,
      'url': url,
      if (thumbnailUrl != null) 'thumbnailUrl': thumbnailUrl,
      if (fileName != null) 'fileName': fileName,
      if (caption != null) 'caption': caption,
      'sortOrder': sortOrder,
      'isPrimary': isPrimary,
      if (uploadedBy != null) 'uploadedBy': uploadedBy,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
